sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/CalciteFrameworkConfigFactory.java

```diff
@@ -19,45 +19,25 @@
 
 import com.google.common.collect.ImmutableList;
 import org.apache.beam.sdk.extensions.sql.impl.planner.BeamRuleSets;
-import org.apache.beam.sdk.extensions.sql.impl.rel.BeamLogicalConvention;
-import org.apache.beam.sdk.extensions.sql.impl.rel.BeamRelNode;
 import org.apache.calcite.config.CalciteConnectionConfig;
 import org.apache.calcite.jdbc.CalciteConnection;
 import org.apache.calcite.jdbc.CalciteSchema;
 import org.apache.calcite.plan.Contexts;
 import org.apache.calcite.plan.ConventionTraitDef;
-import org.apache.calcite.plan.RelOptPlanner.CannotPlanException;
-import org.apache.calcite.plan.RelOptUtil;
 import org.apache.calcite.plan.RelTraitDef;
-import org.apache.calcite.plan.RelTraitSet;
 import org.apache.calcite.prepare.CalciteCatalogReader;
-import org.apache.calcite.rel.RelRoot;
 import org.apache.calcite.schema.SchemaPlus;
-import org.apache.calcite.sql.SqlNode;
 import org.apache.calcite.sql.SqlOperatorTable;
 import org.apache.calcite.sql.fun.SqlStdOperatorTable;
-import org.apache.calcite.sql.parser.SqlParseException;
 import org.apache.calcite.sql.parser.SqlParser;
 import org.apache.calcite.sql.parser.SqlParserImplFactory;
 import org.apache.calcite.sql.util.ChainedSqlOperatorTable;
 import org.apache.calcite.tools.FrameworkConfig;
 import org.apache.calcite.tools.Frameworks;
-import org.apache.calcite.tools.Planner;
-import org.apache.calcite.tools.RelConversionException;
-import org.apache.calcite.tools.ValidationException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-/**
- * The core component to handle through a SQL statement, from explain execution plan, to generate a
- * Beam pipeline.
- */
-class BeamQueryPlanner {
-  private static final Logger LOG = LoggerFactory.getLogger(BeamQueryPlanner.class);
-
-  private final FrameworkConfig config;
-
-  BeamQueryPlanner(CalciteConnection connection) {
+/** Factory that creates {@link FrameworkConfig}. */
+class CalciteFrameworkConfigFactory {
+  public static FrameworkConfig create(CalciteConnection connection) {
```

Comment:
This function only used in `BeamSqlEnv`, might that be a better home for it? (If not, please drop the `public`.)

Reply:
The point I had in mind was to extract boilerplate configuration code out. The BeamSqlEnv is an adaptor for the Calcite planner, that's its main business logic. Creating the configuration from the connection doesn't necessarily seem like what it should be doing.